tests/std/tests/GH_000431_copy_move_family/test.cpp

```diff
@@ -641,6 +642,85 @@ void test_algorithms(CopyFn copy_fn) {
     }
 }
 
+#if _HAS_CXX20
+struct MyIterator { // A contiguous iterator with a weirdly narrow difference type
+    using iterator_concept = std::contiguous_iterator_tag;
+    using value_type       = int;
+    using difference_type  = short;
+
+    int* ptr = nullptr;
+
+    MyIterator() = default;
+    explicit MyIterator(int* p) : ptr{p} {}
+
+    value_type& operator*() const {
+        return *ptr;
+    }
+    value_type* operator->() const {
+        return ptr;
+    }
+
+    MyIterator& operator++() {
+        ++ptr;
+        return *this;
+    }
+    MyIterator operator++(int) {
+        auto tmp = *this;
+        ++ptr;
+        return tmp;
+    }
+
+    MyIterator& operator--() {
+        --ptr;
+        return *this;
+    }
+    MyIterator operator--(int) {
+        auto tmp = *this;
+        --ptr;
+        return tmp;
+    }
+
+    bool operator==(const MyIterator&) const  = default;
+    auto operator<=>(const MyIterator&) const = default;
+
+    value_type& operator[](difference_type i) const {
+        return ptr[i];

```

Comment:
Extreme nitpick: All of your other indices are `n`, and below you use `i` for an iterator. This one would be more consistent if named `n`.